### DISCUSSION TITLE
Using web calendar within an existing website

### DISCUSSION THREAD

**Discussion by AlbaVore** [craigk5n/webcalendar#354] [Q&A] [answered]
I would like to have webcalendar within a page of an existing website. I know that the idea is that webcalendar is intended as a standalone application. However I would like to see if there is away to have it load within a html website via possibly an iFrame or suchlike.
It did work before for me but it appears not to now for some reason.

I would like to have the header and footer of the existing website in place above and below the webcalendar installation. I understand that these can be added as a custom header somewhere so maybe this is a possibility. Has anyone tried this and achieved a tidy presentation o their calendar with such a custom header and footer?

Thanks for any helpful insights.

**Comment by grzchr15** [comment]
i patched reports.php , and edit_reports.php.
So i added type html_part.
And so a simple wordpress plugin seemless adds webcal to this website.
Sorry i did no yet made a pull request

**Comment by AlbaVore** [comment]
Hello, and thanks for your response. However I do not fully understand what it is you have done. Would it be possible to outline this in a bit more detail please as it sounds of interest.

Thanks again.

**Reply by AlbaVore** [reply]
Hello again, Its been a while but I am coming back to this project an am looking for ways to incorporate a webcalendar into a page on a wordpress site. The webcalendar and the wordpress site are on the same domain/directory and the usual iframe method of inserting or the <embed> do not appear to work. I am basically embedding a page within another page in order to have the same header. The alternative is to add the website header to the webcalendar which I am unsure if is possible. Any helpful insights would be much appreciated.

**Reply by AlbaVore** [reply]
Just wondering if anyone is able tom assist in this task? Customising the setup to include  the header and footer of the existing website so that the webcalendar is integrated into the overall website.

**Comment by craigk5n** [comment, marked as answer]
Hi @AlbaVore — a few options for this:

**Custom header/footer in standalone WebCalendar**

WebCalendar reads `includes/custom-header.php` and `includes/custom-footer.php` on every page. Drop your site's header/footer markup into those files and the calendar will render between them — no iframe, no patching, links and forms inside the calendar keep working.

**HTML Reports (what @grzchr15 was describing)**

Admin → Reports lets you define HTML templates that render an event list with no surrounding chrome. The report URL is stable, so you can embed it or paste the output into a static page. Good for read-only views.

**iframe**

Still works. If it stopped, check `X-Frame-Options` / `Content-Security-Policy: frame-ancestors` on the host server, and watch out for cross-site cookie restrictions (SameSite) breaking login state.

**WordPress plugin (new!)**

If your "existing website" is WordPress (or you're open to moving to WordPress), I recently released a from-scratch WebCalendar plugin for WordPress. Drop the `[webcalendar]` shortcode or a Gutenberg block onto any page and the calendar renders inside your theme — header, footer, sidebar, all preserved. No iframe, no `custom-header.php` gymnastics, and it uses your WordPress user accounts.

- Listing: https://wordpress.org/plugins/agenticdaisy-calendar/
- Discussion / testing call: #645

It's a clean re-implementation, not a port, so feature parity isn't 100% — but for "embed a calendar in a page" it's much smoother than patching the standalone version.
